### ISSUE TITLE
Request support for Connect 5W Smart RGB Downlight GU5.3

### ISSUE THREAD

**Issue by BenJamesAndo** [make-all/tuya-local#852]
# Log Message

I've tried with IP of auto or putting the actual IP address in. I either get no log, or the below log. When I add it, it selects RGBCW Lightbulb. Using RGBCW Lightbulb preset it does turn on/off and change colours. Although sometime I have to press the colour twice for it to register. The scenes and timer doesn't work.
```
Logger: homeassistant.util.async_
Source: util/async_.py:166
First occurred: 9:13:57 AM (2 occurrences)
Last logged: 9:19:53 AM

Detected blocking call to sleep inside the event loop. This is causing stability issues. Please report issue to the custom integration author for tuya_local doing blocking calls at custom_components/tuya_local/device.py, line 78: self._api = tinytuya.Device(dev_id, address, local_key)
```

# Information about DPS mappings

```
{
  "result": {
    "properties": [
      {
        "code": "switch_led",
        "custom_name": "",
        "dp_id": 20,
        "time": 1687994470663,
        "value": false
      },
      {
        "code": "work_mode",
        "custom_name": "",
        "dp_id": 21,
        "time": 1687994470663,
        "value": "colour"
      },
      {
        "code": "bright_value",
        "custom_name": "",
        "dp_id": 22,
        "time": 1687994470663,
        "value": 1000
      },
      {
        "code": "temp_value",
        "custom_name": "",
        "dp_id": 23,
        "time": 1687994470663,
        "value": 225
      },
      {
        "code": "colour_data",
        "custom_name": "",
        "dp_id": 24,
        "time": 1687994470663,
        "value": "007b03b603e8"
      },
      {
        "code": "scene_data",
        "custom_name": "",
        "dp_id": 25,
        "time": 1687994470663,
        "value": "07464602000003e803e800000000464602007803e803e80000000046460200f003e803e800000000464602003d03e803e80000000046460200ae03e803e800000000464602011303e803e800000000"
      },
      {
        "code": "countdown",
        "custom_name": "",
        "dp_id": 26,
        "time": 1687994470663,
        "value": 0
      },
      {
        "code": "music_data",
        "custom_name": "",
        "dp_id": 27,
        "time": 1687059394486,
        "value": ""
      },
      {
        "code": "control_data",
        "custom_name": "",
        "dp_id": 28,
        "time": 1687059394486,
        "value": ""
      },
      {
        "code": "rhythm_mode",
        "custom_name": "",
        "dp_id": 30,
        "time": 1687951002499,
        "value": "AAAAfwQBBh4AAAAAFBkBCh4AAAAAZGQBFAAAAAAAUDIBFwAAAAAABQA="
      },
      {
        "code": "sleep_mode",
        "custom_name": "",
        "dp_id": 31,
        "time": 1687951002851,
        "value": "AAA="
      },
      {
        "code": "wakeup_mode",
        "custom_name": "",
        "dp_id": 32,
        "time": 1687951003207,
        "value": "AAA="
      }
    ]
  },
  "success": true,
  "t": 1687994669479,
  "tid": "ee0c06dd160a11ee97740a4306e7517e"
}
```

# Product ID
`3cmiojdcymzcuod0`

# Information about how the device functions
Fairly standard Tuya smart RGB light. 
https://connectsmarthome.com.au/product/connect-5w-smart-rgb-downlight-gu5-3
https://www.arrowcomputers.com.au/shop/product/csh-gu53rgb5w-connect-5w-smart-rgb-downlight-gu5-3-12744
Model CSH-GU53RGB5W
Brand: Connect Smarthome by Laser Co
Manual: https://www.connectsmarthome.com.au/manual/lighting/Manual%20-%20CSH%20RGB%20Smart%20Light.pdf
Tuya API Explorer calls it GU5.3 RGB Light

**Comment by make-all** [comment]
If you managed to add it, there should've been a log message with the dps listed unless it was a perfect match for RGBCW lightbulb, which it seems not. Or, now that it is added, you can download the device diagnostics from Settings /Devices to get the dps listing.

**Comment by BenJamesAndo** [comment]
Oh gotcha. Below is the diagnostics.

```
 "integration_manifest": {
    "domain": "tuya_local",
    "name": "Tuya Local",
    "codeowners": [
      "@make-all"
    ],
    "config_flow": true,
    "dependencies": [],
    "documentation": "https://github.com/make-all/tuya-local",
    "integration_type": "device",
    "iot_class": "local_push",
    "issue_tracker": "https://github.com/make-all/tuya-local/issues",
    "requirements": [
      "pycryptodome~=3.18",
      "tinytuya==1.12.8"
    ],
    "version": "2023.6.1",
    "is_built_in": false
  },
  "data": {
    "name": "RGBCW Lightbulb",
    "type": "rgbcw_lightbulb",
    "device_id": "**REDACTED**",
    "device_cid": "",
    "local_key": "**REDACTED**",
    "host": "**REDACTED**",
    "protocol_version": 3.3,
    "tinytuya_version": "1.12.8",
    "api_version_set": 3.3,
    "api_version_used": 3.3,
    "api_working": true,
    "status": {},
    "cached_state": {
      "updated_at": 1688004239.8565361,
      "20": false,
      "21": "colour",
      "22": 1000,
      "23": 225,
      "24": "007b03b603e8",
      "25": "07464602000003e803e800000000464602007803e803e80000000046460200f003e803e800000000464602003d03e803e80000000046460200ae03e803e800000000464602011303e803e800000000",
      "26": 0
    },
    "pending_state": {},
    "connected": true,
    "force_dps": [],
    "home_assistant": {
      "name": "RGBCW Lightbulb",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "select.rgbcw_lightbulb_scene",
            "state": "Gorgeous",
            "attributes": {
              "options": [
                "Night",
                "Read",
                "Working",
                "Leisure",
                "Soft",
                "Colorful",
                "Dazzling",
                "Gorgeous"
              ],
              "icon": "mdi:palette",
              "friendly_name": "RGBCW Lightbulb Scene"
            },
            "last_changed": "2023-06-29T01:58:20.039587+00:00",
            "last_updated": "2023-06-29T01:58:20.039587+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "light.rgbcw_lightbulb",
            "state": "off",
            "attributes": {
              "min_color_temp_kelvin": 2000,
              "max_color_temp_kelvin": 6535,
              "min_mireds": 153,
              "max_mireds": 500,
              "effect_list": [
                "Scene",
                "Music"
              ],
              "supported_color_modes": [
                "color_temp",
                "hs"
              ],
              "scene_data": "07464602000003e803e800000000464602007803e803e80000000046460200f003e803e800000000464602003d03e803e80000000046460200ae03e803e800000000464602011303e803e800000000",
              "friendly_name": "RGBCW Lightbulb",
              "supported_features": 4
            },
            "last_changed": "2023-06-29T01:58:20.039918+00:00",
            "last_updated": "2023-06-29T01:58:20.039918+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "config",
          "device_class": null,
          "original_device_class": null,
          "icon": null,
          "unit_of_measurement": "min",
          "state": {
            "entity_id": "number.rgbcw_lightbulb_timer",
            "state": "0.0",
            "attributes": {
              "min": 0.0,
              "max": 1440.0,
              "step": 1.0,
              "mode": "auto",
              "unit_of_measurement": "min",
              "icon": "mdi:timer",
              "friendly_name": "RGBCW Lightbulb Timer"
            },
            "last_changed": "2023-06-29T01:58:20.039367+00:00",
            "last_updated": "2023-06-29T01:58:20.039367+00:00"
          }
        }
      ]
    }
  }
}
```

**Comment by make-all** [comment]
It does look to be exactly the same as RGBCW lightbulb, except perhaps the scene format may be different.  Or you may need to change the lightbulb effect to scene as well as setting your scene with the separate scene select entity to get them to work.  Since it is a fixed list of scenes, based on one specific lightbulbs predefined scenes, the support for scenes will be limited anyway.

**Comment by BenJamesAndo** [comment]
Thanks for the insights. You're completely right. If I set both the scene select entity and scene effect then the scenes work. I woudn't have thought of that, thanks.

I did some more tests with the timer and turns out it does work perfectly. I think I must have tested it with local tuya and forgot and didn't bother to test it again with tuya local. Thanks for your help, and sorry for the hassle. Since nothing further is needed I'll close this issue.
